File: star1/eto1.rb
year = 2000 # グレゴリオ暦年

# if 式は 条件分岐のための基本です。
# 条件が単純な場合には switch case 式を使うと、
# すっきり書くことができます。

# 12で割った余りに応じて、干支を表示します。
case year % 12
when 0
  puts "申年"
when 1
  puts "酉年"
when 2
  puts "戌年"
when 3
  puts "亥年"
when 4
  puts "子年"
when 5
  puts "丑年"
when 6
  puts "寅年"
when 7
  puts "卯年"
when 8
  puts "辰年"
when 9
  puts "巳年"
when 10
  puts "午年"
when 11
  puts "未年"
end

File: star1/eto1.c
#include <stdio.h>  // 標準入出力関数

int main(int argc, char const *argv[]) {
  int year = 2000;  // グレゴリオ暦年

  // if 文は 条件分岐のための基本です。
  // 条件が単純な場合には switch case 文を使うと、
  // すっきり書くことができます。

  // 12で割った余りに応じて、干支を表示します。
  switch (year % 12) {
  case 0:
    printf("申年\n");
    break;
  case 1:
    printf("酉年\n");
    break;
  case 2:
    printf("戌年\n");
    break;
  case 3:
    printf("亥年\n");
    break;
  case 4:
    printf("子年\n");
    break;
  case 5:
    printf("丑年\n");
    break;
  case 6:
    printf("寅年\n");
    break;
  case 7:
    printf("卯年\n");
    break;
  case 8:
    printf("辰年\n");
    break;
  case 9:
    printf("巳年\n");
    break;
  case 10:
    printf("午年\n");
    break;
  case 11:
    printf("未年\n");
    break;
  }

  return 0;
}
